floater/generators.py

```diff
@@ -331,3 +333,34 @@ def subset_floats_from_mask(self, xx, yy):
         return floats_ocean
 
 
+    def npart_to_2D_array(self, var, ds1d):
+        """Constructs 2d dataset from 1d dataset of a physical variable.
+
+        PARAMETERS
+        ----------
+        var : str
+            The name of the physical variable
+        ds1d : 1d dataset
+            The 1d dataset of the physical variable with dimension 'npart'
+
+        RETURNS
+        -------
+        ds2d : 2d dataset
+            The 2d dataset of the physical variable with dimensions 'lat' and 'lon'
+        """
+
+        Nx = self.Nx
+        Ny = self.Ny
+        Nt = Nx*Ny
+        frame = pd.DataFrame({'index': range(1, Nt+1), var: np.zeros(Nt)})
+        framei = frame.set_index('index')
+        df = ds1d.to_dataframe()
+        ocean_bools = self.ocean_bools
+        framei.loc[ocean_bools==True, var] = df[var].values.astype(np.float32)
+        framei.loc[ocean_bools==False, var] = np.float32('nan')
```

Comment:
Here I think you could just leave off `var` in order to apply to all variables in the dataset, no?